ci/ray_ci/linux_container.py

```diff
@@ -0,0 +1,70 @@
+import os
+import subprocess
+import sys
+from typing import List, Optional
+
+from ci.ray_ci.container import Container, _DOCKER_CAP_ADD
+
+
+class LinuxContainer(Container):
+    def install_ray(self, build_type: Optional[str] = None) -> List[str]:
+        env = os.environ.copy()
+        env["DOCKER_BUILDKIT"] = "1"
+        subprocess.check_call(
+            [
+                "docker",
+                "build",
+                "--pull",
+                "--build-arg",
+                f"BASE_IMAGE={self._get_docker_image()}",
+                "--build-arg",
+                f"BUILD_TYPE={build_type or ''}",
+                "-t",
+                self._get_docker_image(),
+                "-f",
+                "/ray/ci/ray_ci/tests.env.Dockerfile",
+                "/ray",
+            ],
+            env=env,
+            stdout=sys.stdout,
+            stderr=sys.stderr,
+        )
+
+    def _get_run_command(
+        self,
+        script: List[str],
+        gpu_ids: Optional[List[int]] = None,
+    ) -> List[str]:
+        command = [
+            "docker",
+            "run",
+            "-i",
+            "--rm",
+            "--env",
+            "NVIDIA_DISABLE_REQUIRE=1",
+            "--volume",
+            "/tmp/artifacts:/artifact-mount",
+            "--add-host",
+            "rayci.localhost:host-gateway",
+        ]
+        for volume in self.volumes:
```

Comment:
as discussed, please add get functions for these fields referenced from `self`, or send them in as an arg.

you can also have a `get` function that returns a struct that has volume, env, docker images all at the same time or something.

I am not particularly worried about if the returned stuff has read-only protection or not.

the main thing I would like to have is to have a clear set of interfaces that the child class can use from the abstract parent class.